vicon_object_server_library/src/vicon_object_server_client.cpp

```
@@ -0,0 +1,429 @@
+    
+    geometry_msgs::Pose waypoint;
+    tf::poseEigenToMsg(trajAddWaypoint,waypoint);
+    trajAddWaypointSrv.request.waypoint = waypoint;
+
+    if(clients_["TrajAddWaypoint"].call(trajAddWaypointSrv))
+    {
+        int status(trajAddWaypointSrv.response.status);
+        std::string message(trajAddWaypointSrv.response.message.c_str());
+
+        VICONObjectClient::printStatus(status,message,"TrajAddWaypoint");
+
+        return status;
+    } 
+    else 
+    {
+        ROS_ERROR("VICON Object Client: Failed to call service TrajAdaWaypoint");
+
+        return 3;
+    }
+    
+}
+int VICONObjectClient::trajDump(const std::string &name,const std::string &file)
+{
+    TrajDump trajDumpSrv;
+    trajDumpSrv.request.name =name;
+    trajDumpSrv.request.file =file;
+
+    if(clients_["TrajDump"].call(trajDumpSrv))
+    {
+        int status(trajDumpSrv.response.status);
+        std::string message(trajDumpSrv.response.message.c_str());
+
+        VICONObjectClient::printStatus(status,message,"TrajDump");
+
+        return status;
+    } 
+    else 
+    {
+        ROS_ERROR("VICON Object Client: Failed to call service TrajDump");
+        return 3;
+    }
+}
+Eigen::Affine3d*  VICONObjectClient::trajGetWaypoint(std::string& trajectory, int index)
+{
+    TrajGetWaypoint trajGetWaypointSrv;
+    trajGetWaypointSrv.request.trajectory = trajectory;
+    trajGetWaypointSrv.request.index = index;
+
+    if(clients_["TrajGetWaypoint"].call(trajGetWaypointSrv))
+    {
+        int status(trajGetWaypointSrv.response.status);
+        std::string message(trajGetWaypointSrv.response.message.c_str());
+        Eigen::Affine3d* pose = new Eigen::Affine3d;
+
+        tf::poseMsgToEigen(trajGetWaypointSrv.response.pose, *pose);
+        VICONObjectClient::printStatus(status,message,"TrajGetWaypoint");
+
+        return pose;
+    } 
+    else 
+    {
+        ROS_ERROR("VICON Object Client: Failed to call service TrajGet");
+
+        return NULL;
```

Hm. I'm paranoid about returning NULL, but don't have a better solution. Maybe throw an exception?

---

This function should just return `Eigen::Affine3d` and throw an exception on an error.